include/deal.II/base/memory_space_data.h

```diff
@@ -25,12 +25,124 @@
 #include <functional>
 #include <memory>
 
+#ifdef DEAL_II_USE_KOKKOS_BACKEND
+#  include <Kokkos_Core.hpp>
+#endif
+
 DEAL_II_NAMESPACE_OPEN
 
 /**
  */
 namespace MemorySpace
 {
+#ifdef DEAL_II_USE_KOKKOS_BACKEND
+  /**
+   * Data structure
+   */
+  template <typename Number, typename MemorySpace>
+  struct MemorySpaceData
+  {
+    MemorySpaceData()
+      : values("memoryspace data", 0)
+    {}
+
+    bool
+    has_data()
+    {
+      return values.extent(0) > 0;
```

Comment:
is there always at least one extent? how does a default-constructed object look like?

Reply:
A default-constructed object has all (dynamic) extents 0.